src/wasm/wasm-validator.cpp

```diff
@@ -2030,6 +2031,9 @@ void FunctionValidator::visitRefFunc(RefFunc* curr) {
   }
   auto* func = getModule()->getFunctionOrNull(curr->func);
   shouldBeTrue(!!func, curr, "function argument of ref.func must exist");
+  shouldBeTrue(curr->type.isFunction(),
+               curr,
+               "ref.func must have a function reference type");
```

Comment:
IIRC, the type should also not be nullable. It would be good to check that here.

Reply:
Adding, thanks.